src/main/java/com/hbm/inventory/container/ContainerBase.java

```diff
@@ -0,0 +1,67 @@
+package com.hbm.inventory.container;
+
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.entity.player.InventoryPlayer;
+import net.minecraft.inventory.Container;
+import net.minecraft.inventory.IInventory;
+import net.minecraft.inventory.Slot;
+import net.minecraft.item.ItemStack;
+
+/**For now, only used for stuff with filters and crates as a reference implementation,
+ * because I really needed to get the te from a container
+ * But you should very much use this to kill the giant amount of boilerplate in container classes
+ * @author 70k **/
+public class ContainerBase extends Container {
```

Comment:
If you want to make this even more useful and not have to worry about adding a constructor (since containers will generally do their own shit), make this an abstract class. This can also force implementers to do certain things so they don't shoot they own foot.

Reply:
oh shit right I honestly forgot abstracts exist
the problem is
I NEED the constructor so I can acess the TE from the container